From 1abf88122eca0bbdb163fb1c5346fcb7514f4e40 Mon Sep 17 00:00:00 2001
From: Kihwal Lee <kihwal@apache.org>
Date: Thu, 20 Oct 2016 11:09:20 -0500
Subject: [PATCH 2077/2748] HDFS-10627. Volume Scanner marks a block as
 "suspect" even if the exception is
 network-related. Contributed by Rushabh S Shah.
 Updated CHANGES.txt (cherry picked from commit
 5c0bffddc0cb824a8a2751bcd0dc3e15ce081727)

(cherry picked from commit 6ae7b9f800f6d349cfeef50950d4eb72d1c1709f)

 Conflicts:
	hadoop-hdfs-project/hadoop-hdfs/CHANGES.txt

Change-Id: I05ffa5e4a98895968464050093503269dab13828
---
 .../hadoop/hdfs/server/datanode/BlockSender.java   |    4 ++--
 1 file changed, 2 insertions(+), 2 deletions(-)

diff --git a/hadoop-hdfs-project/hadoop-hdfs/src/main/java/org/apache/hadoop/hdfs/server/datanode/BlockSender.java b/hadoop-hdfs-project/hadoop-hdfs/src/main/java/org/apache/hadoop/hdfs/server/datanode/BlockSender.java
index 892c852..4d9b200 100644
--- a/hadoop-hdfs-project/hadoop-hdfs/src/main/java/org/apache/hadoop/hdfs/server/datanode/BlockSender.java
+++ b/hadoop-hdfs-project/hadoop-hdfs/src/main/java/org/apache/hadoop/hdfs/server/datanode/BlockSender.java
@@ -612,10 +612,10 @@ private int sendPacket(ByteBuffer pkt, int maxChunks, OutputStream out,
         String ioem = e.getMessage();
         if (!ioem.startsWith("Broken pipe") && !ioem.startsWith("Connection reset")) {
           LOG.error("BlockSender.sendChunks() exception: ", e);
-        }
-        datanode.getBlockScanner().markSuspectBlock(
+          datanode.getBlockScanner().markSuspectBlock(
               volumeRef.getVolume().getStorageID(),
               block);
+        }
       }
       throw ioeToSocketException(e);
     }
-- 
1.7.9.5

